packages/controller-utils/src/util.ts

```diff
@@ -251,20 +251,45 @@ export async function safelyExecuteWithTimeout<Result>(
 }
 
 /**
- * Convert an address to a checksummed hexidecimal address.
+ * Convert an address to a checksummed hexadecimal address.
  *
  * @param address - The address to convert.
- * @returns A 0x-prefixed hexidecimal checksummed address, if address is valid. Otherwise original input 0x-prefixe, if address is valid. Otherwise original input 0x-prefixed.
+ * @returns The address in 0x-prefixed hexadecimal checksummed form if it is valid.
  */
-export function toChecksumHexAddress(address: string) {
+export function toChecksumHexAddress(address: string): string;
```

Comment:
Good catch!

Reply:
After I made that suggestion I looked back and realized that it didn't return `Hex` before, it always returned `string` before, regardless of the input. `Hex` does make sense, but if we changed that, it would be breaking, I believe. Does that sound right?